lib/widgets/sticky_header/sticky_header_render.dart

```diff
@@ -0,0 +1,202 @@
+    RenderStickyHeaderCallback? callback,
+    bool overlapHeaders: false,
+    RenderBox? header,
+    RenderBox? content,
+  })  : _scrollPosition = scrollPosition,
+        _callback = callback,
+        _overlapHeaders = overlapHeaders {
+    if (content != null) add(content);
+    if (header != null) add(header);
+  }
+
+  set scrollPosition(ScrollPosition newValue) {
+    if (_scrollPosition == newValue) {
+      return;
+    }
+    final ScrollPosition oldValue = _scrollPosition;
+    _scrollPosition = newValue;
+
+    markNeedsLayout();
+    if (attached) {
+      oldValue.removeListener(markNeedsLayout);
+      newValue.addListener(markNeedsLayout);
+    }
+  }
+
+  set callback(RenderStickyHeaderCallback? newValue) {
+    if (_callback == newValue) {
+      return;
+    }
+    _callback = newValue;
+    markNeedsLayout();
+  }
+
+  set overlapHeaders(bool newValue) {
+    if (_overlapHeaders == newValue) {
+      return;
+    }
+    _overlapHeaders = newValue;
+    markNeedsLayout();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    _scrollPosition.addListener(markNeedsLayout);
+  }
+
+  @override
+  void detach() {
+    _scrollPosition.removeListener(markNeedsLayout);
+    super.detach();
+  }
+
+  // short-hand to access the child RenderObjects
+  RenderBox get _headerBox => lastChild!;
+
+  RenderBox get _contentBox => firstChild!;
+
+  // Put header at its original position
+  void resetHeaderPosition() {
+    final headerParentData = _headerBox.parentData as MultiChildLayoutParentData;
+    headerParentData.offset = const Offset(0.0, 0.0);
+  }
+
+  @override
+  void performLayout() {
+    // ensure we have header and content boxes
+    assert(childCount == 2);
+
+    // layout both header and content widget
+    final childConstraints = constraints.loosen();
+    _headerBox.layout(childConstraints, parentUsesSize: true);
+    _contentBox.layout(childConstraints, parentUsesSize: true);
+
+    final headerHeight = _headerBox.size.height;
+    final contentHeight = _contentBox.size.height;
+
+    // determine size of ourselves based on content widget
+    final width = max(constraints.minWidth, _contentBox.size.width);
+    final height =
+        max(constraints.minHeight, _overlapHeaders ? contentHeight : headerHeight + contentHeight);
+    size = Size(width, height);
+    assert(size.width == constraints.constrainWidth(width));
+    assert(size.height == constraints.constrainHeight(height));
+    assert(size.isFinite);
+
+    // place content underneath header
+    final contentParentData = _contentBox.parentData as MultiChildLayoutParentData;
+    contentParentData.offset = Offset(0.0, _overlapHeaders ? 0.0 : headerHeight);
+
+    // determine by how much the header should be stuck to the top
+    final double stuckOffset = determineStuckOffset();
+
+    if (_scrollPosition.userScrollDirection == ScrollDirection.idle) {
+      resetHeaderPosition();

```

Comment:
```suggestion
 Future.delayed(const Duration(milliseconds: 5000), () {
      resetHeaderPosition();
});
```

@nqhhdev timer is not working here (resetHeaderPosition is not called), I still need to find a way to setup a deboucetime for this but since it's in a performLayout function (called many times) I don't know how to do it with a MultiChildLayoutDelegate class